client/my-sites/site-settings/redirect-to-general.jsx

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import page from 'page';
+import { flowRight } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { isSiteAutomatedTransfer } from 'state/selectors';
+
+const redirectNonJetpackToGeneral = ( ComponentToRender ) => {
+	class RedirectNonJetpackToGeneral extends Component {
+		static propTypes = {
+			siteSlug: PropTypes.string,
+		}
+
+		componentDidMount() {
+			this.verifySiteIsJetpack();
+		}
+
+		componentDidUpdate() {
+			this.verifySiteIsJetpack();
+		}
+
+		verifySiteIsJetpack() {
+			if ( this.props.siteIsJetpack === false || this.props.siteIsAtomic ) {
+				this.redirectToGeneral();
+			}
+		}
+
+		redirectToGeneral = () => {
+			const {
+				siteSlug,
+			} = this.props;
+
+			if ( ! siteSlug ) {
+				return null;
+			}
+			page( '/settings/general/' + siteSlug );
```

Comment:
An architectural question for you to consider: this component would be much more reusable if we're able to alter the route we redirect to. By default it could redirect to `/settings/general/:siteSlug`, but it would allow that to be overriden by a custom, optional prop. That change would also probably involve changing the name of the HoC, too. What do you think?

Reply:
@tyxla, true, this is a great idea :) I initially implemented it with a url passed as a prop and a default fallback to the login link ```redirectTo: getRedirectTo(state)```. 
Then I considered the following: 
- the conditions for which the redirection happens are pretty specific (non-JP + Atomic), which possibly limits the use cases anyway, so no need to be extra flexible 
-  it makes sense to have a standard place to redirect to, which would `enforces` a bit thinking about that aspect: what do we want to happen for that specific user group.

But yes, this is very much a possibility and a fast edit. I have no strong preferences and happy to do it. What do you think @sirreal?

Reply:
It's a good thing to consider and I'm glad to hear you already have. If we don't have a specific use case, I think it's OK to pass on this. It can be modified when the need arises.

Reply:
UPDATE: it is included now